107/src/main/java/org/ehcache/jsr107/Eh107MXBean.java

```diff
@@ -28,17 +28,16 @@
 
   private final ObjectName objectName;
 
-  Eh107MXBean(String cacheName, Eh107CacheManager cacheManager, final String beanName) {
-    this.objectName = createObjectName(cacheName, cacheManager, beanName);
+  Eh107MXBean(String cacheName, URI cacheManagerURI, final String beanName) {
+    this.objectName = createObjectName(cacheName, cacheManagerURI, beanName);
   }
 
   private String sanitize(String string) {
     return string == null ? "" : string.replaceAll(",|:|=|\n", ".");
   }
 
-  private ObjectName createObjectName(String cacheName, Eh107CacheManager cacheManager, String beanName) {
-    URI uri = cacheManager.getURI();
-    String cacheManagerName = sanitize(uri != null ? uri.toString() : "null");
+  private ObjectName createObjectName(String cacheName, URI cacheManagerURI, String beanName) {
+    String cacheManagerName = sanitize(cacheManagerURI != null ? cacheManagerURI.toString() : "null");
```

Comment:
I just been wondering. Can the cache manager URI really be null? Or it should be a precondition that it isn't?